akka-actor/src/main/scala/akka/pattern/AskSupport.scala

```diff
@@ -257,6 +266,36 @@ object AskableActorRef {
    */
   private[pattern] def $qmark$extension(actorRef: ActorRef, message: Any, timeout: Timeout): Future[Any] =
     actorRef.internalAsk(message, timeout, ActorRef.noSender)
+
+  private def messagePartOfException(message: Any, sender: ActorRef): String = {
+    val msg = if (message == null) "unknown" else message
+    val wasSentBy = if (sender == ActorRef.noSender) "" else s" was sent by [$sender]"
+    s"Message of type [${msg.getClass.getName}]$wasSentBy."
```

Comment:
Would it be possible to add customization for how the message is logged? One annoying thing in Lagom is that `msg.getClass.getName` is always `com.lightbend.lagom.scaladsl.persistence.CommandEnvelope` because of the wrapper that it uses to include the persistence ID with the message payload.

Maybe `ask` could take an optional `logMessage: () => String` parameter, or use a typeclass (for convenience with `?` syntax).

Reply:
In Typed `EntityRef` we unwrap that. In Lagom we should `ecover` the `Future` to create a custom `AskTimeOutException` with a message that makes sense.